packages/grid/_modules_/grid/components/ErrorOverlay.tsx

```diff
@@ -1,12 +1,21 @@
 import * as React from 'react';
 import { GridApiContext } from './GridApiContext';
-import { GridOverlay } from './containers/GridOverlay';
+import { GridOverlay, GridOverlayProps } from './containers/GridOverlay';
 
-export interface ErrorOverlayProps {
+export interface ErrorOverlayProps extends GridOverlayProps {
   message?: string;
 }
-export function ErrorOverlay({ message }: ErrorOverlayProps) {
-  const apiRef = React.useContext(GridApiContext);
-  const defaultLabel = apiRef!.current.getLocaleText('errorOverlayDefaultLabel');
-  return <GridOverlay>{message || defaultLabel}</GridOverlay>;
-}
+
+export const ErrorOverlay = React.forwardRef<HTMLDivElement, ErrorOverlayProps>(
+  function ErrorOverlay(props: ErrorOverlayProps, ref) {
+    const { message, ...other } = props;
+    const apiRef = React.useContext(GridApiContext);
+    const defaultLabel = apiRef!.current.getLocaleText('errorOverlayDefaultLabel');
+
+    return (
+      <GridOverlay ref={ref} {...other}>
```

Comment:
the issue with the rootelement might come from here, as there are more than just a message in the props

Reply:
I was under the impression that root element is it the tip of the iceberg, that it surface that we need to destructure all the extra "base props" as developers would need too when customizing the components. For instance, the `rootElement` prop shouldn't be applied on the root element, but an aria-label/onClick/etc. might.

Reply:
yes I need to strip the `GridBaseComponentProps`. I'm currently working on a solution for that ;)

Reply:
Ok I thought about how to fix that problem and I came up with a helper hook that strips the props from the `GridBaseComponentProps` and what is left is spread on the component.